### DISCUSSION TITLE
Is it really necessary to [[nodiscard]] the return value of the write() method?

### DISCUSSION THREAD

**Discussion by scalpel4k** [stephenberry/glaze#1104] [Q&A] [answered]
While refactoring our code I found a few places where I don't care whether the writing was successful, but the linter now generates a bunch of warnings .... I wonder if the library should really care about the usage of the return value. What do you think?

**Comment by stephenberry** [comment, marked as answer]
There are a few ways to clean up the code when you don't care about successful writing:
- Use `std::ignore`
```c++
std::ignore = glz::write_json(value, buffer);
```
This is probably what you want, albeit it requires a bit more code from the developer.

- Use the exceptions API.
```c++
#include "glaze/glaze_exceptions.hpp"

glz::ex::write_json(value, buffer);
```
Now, in the unlikely case of an error you'll get an exception, but you don't have to handle it here.

- Cast to void
```c++
(void)glz::write_json(value, buffer);
```
I would prefer `std::ignore` because it is more obvious what is going on, but this is an alternative approach.

**Reply by scalpel4k** [reply]
thx for your answer. maybe my approach to using [[nodiscard]] is a bit different: "don't use this method if you're not using the result, because the method is all about generating this return value, i.e. it's not having any side effects other than generating the return value".
In this case I think, calling the method isn't 'wasted' per se if your not using the return value.

Anyway, you're suggestions definitely work ...

**Reply by stephenberry** [reply]
I agree that the primary use for [[nodiscard]] is when there are no other side effects. But, I also find it helpful for reminding developers that a possible error isn't being handled. I find being explicit about error handling prevents coding mistakes that were relying on valid results.
Thanks for your thoughts! I just wanted to provide my reasoning for the design.

**Reply by scalpel4k** [reply]
ok, the Rust way ;-)
